controllers/applicationvolumereplication_controller.go

```diff
@@ -188,105 +206,330 @@ func getMostRecentConditions(conditions []metav1.Condition) []metav1.Condition {
+		}
+
+		r.Log.Info(fmt.Sprintf("Unpausing subscription %s for new home cluster %s", subscription.Name, newHomeCluster))
+
+		err = r.unpauseSubscription(subscription)
+		if err != nil {
+			r.Log.Error(err, "failed to unpause subscription", "name", subscription.Name)
+
+			return nil, requeue
+		}
+
+		// Subscription has been unpaused. Stop processing it and wait for the next Reconciler iteration
+		r.Log.Info("Subscription unpaused. It will be processed in the next reconciler iteration", "name", subscription.Name)
+
+		return nil, requeue
 	}
-	if err := r.Client.Status().Update(ctx, avr); err != nil {
-		return errorswrapper.Wrap(err, "failed to update AVR status")
+
+	exists, err := r.vrgManifestWorkAlreadyExists(avr, subscription)
+	if err != nil {
+		return nil, requeue
+	}
+
+	if exists {
+		return nil, !requeue
+	}
+	// This subscription is ready for manifest (VRG) creation
+	placementDecision, err := r.processUnpausedSubscription(avr, subscription)
+	if err != nil {
+		r.Log.Error(err, "Failed to process unpaused subscription", "name", subscription.Name)
+
+		return nil, requeue
+	}
+
+	r.Log.Info(fmt.Sprintf("placementDecisions %v - requeue: %t", placementDecision, !requeue))
+
+	return &placementDecision, !requeue
+}
+
+func (r *ApplicationVolumeReplicationReconciler) isSubsriptionPausedForDR(labels map[string]string) bool {
+	return labels != nil &&
+		labels[RamenDRLabelName] != "" &&
+		strings.EqualFold(labels[RamenDRLabelName], "protected") &&
+		labels[subv1.LabelSubscriptionPause] != "" &&
+		strings.EqualFold(labels[subv1.LabelSubscriptionPause], "true")
+}
+
+// processPausedSubscription selects the target cluster from the subscription or
+// from the user selected cluster and restores all PVs (belonging to the subscription)
+// to the target cluster and then it unpauses the subscription
+func (r *ApplicationVolumeReplicationReconciler) processPausedSubscription(
+	avr *ramendrv1alpha1.ApplicationVolumeReplication,
+	subscription *subv1.Subscription) (string, error) {
+	r.Log.Info("Processing paused subscription", "name", subscription.Name)
+
+	// find new home cluster (could be the failover cluster)
+	newHomeCluster := r.findNextHomeCluster(avr, subscription)
+
+	if newHomeCluster == "" {
+		return "", fmt.Errorf("failed to find new home cluster: avr %s, subscription %s", avr.Name, subscription.Name)
+	}
+
+	mw, err := r.findManifestWork(subscription, newHomeCluster, MWTypePV)
+	if err != nil {
+		return "", fmt.Errorf("failed to get PV ManifestWork (%w)", err)
+	}
+
+	if mw != nil {
+		if IsManifestInAppliedState(mw) {
+			r.Log.Info(fmt.Sprintf("ManifestWork applied (%+v)", mw))
+
+			return newHomeCluster, nil
+		}
+
+		return "", fmt.Errorf("ManifestWork has not been applied yet (%+v)", mw)
+	}
+
+	err = r.deleteExistingManfiestWork(avr, subscription)
+	if err != nil {
+		return "", fmt.Errorf("failed to delete existing ManifestWork (%w)", err)
+	}
+
+	err = r.restorePVFromBackup(avr, subscription, newHomeCluster)
```

Comment:
We just created the PV MW, and will now unpause the subscription, should we not wait till the MW reported as applied?

Reply:
Good catch.  That wasn't supposed to happen.  I thought I only unpaused when the manifest is in the applied state (that's the whole point of Travis' work), but it doesn't look like I did.  Fixing it.

Reply:
Fixed in the latest PR